aimon/extensions/react.py

```diff
@@ -0,0 +1,168 @@
+        self.llm_app = llm_app
+        self.context_extractor = context_extractor
+        self.react_configuration = react_configuration
+        self.client = Client(auth_header="Bearer {}".format(self.react_configuration.aimon_api_key))
+
+    def create_payload(self, context, user_query, user_instructions, generated_text):
+        
+        aimon_payload = {
+            'context':context,
+            'user_query':user_query,
+            'generated_text':generated_text,
+            'instructions':user_instructions,
+        }
+
+        aimon_payload['publish'] = self.react_configuration.publish
+        aimon_payload['config'] = { 'hallucination': {'detector_name': 'default'},
+                                    'instruction_adherence': {'detector_name': 'default'},}
+
+        if self.react_configuration.publish:
+            aimon_payload['application_name'] = self.react_configuration.application_name
+            aimon_payload['model_name'] = self.react_configuration.model_name
+
+        return aimon_payload
+
+    def detect_aimon_response(self,aimon_payload):
+        
+        try:
+            detect_response = self.client.inference.detect(body=[aimon_payload])
+            # Check if the response is a list
+            if isinstance(detect_response, list) and len(detect_response) > 0:
+                detect_result = detect_response[0]
+            elif isinstance(detect_response, dict):
+                detect_result = detect_response  # Single dict response
+            else:
+                raise ValueError("Unexpected response format from detect API: {}".format(detect_response))
+        except Exception as e:
+                # Log the error and raise it
+                print(f"Error during detection: {e}")
+                raise
+        
+        return detect_result
+
+    ## ReAct -> Reason and Act
+    def react(self, user_query, user_instructions,):
+        """
+        AIMon-ReAct -> Reason and Act with AIMon
+
+        where ReAct score: 
+            1    if after n attempts of ReAct, the LLM follows instructions and generates a non hallucinated response.
+            0.5  if after n attempts of ReAct, either the LLM response follows user_instructions or is under hallucination threshold [BUT NOT BOTH]. 
+            0    otherwise.
+        """
+        result = {
+            'responses':             [],
+            'hallucination_scores':  [],
+            'adherence':             [],                
+            'react_score':           0     ## 0 by assumption
+        }
+
+        llm_response = self.llm_app(user_query, user_instructions, reprompted_flag=False)
+        
+        context = self.context_extractor(user_query, user_instructions, llm_response)    
+
+        ## Generated text for LLM Response, if the user employs the LlamaIndex framework
+        if llm_response.response or self.react_configuration.framework=="llamaindex":
```

Comment:
flip this to 

```python
self.react_configuration.framework=="llamaindex" or llm_response.response:
    generated_text = llm_response.response
...
```

This way if the framework is not llamaindex then the second statement `lm_response.response` will not get evaluated
```